static/js/browser2/statvar_group_node.tsx

```diff
@@ -0,0 +1,208 @@
+import { StatVarGroupNodeType, StatVarHierarchyNodeType } from "./util";
+import { StatVarNode } from "./statvar_node";
+
+interface StatVarGroupNodePropType {
+  dcid: string;
+  placeName: string;
+  statVarGroupId: string;
+  data: { [key: string]: StatVarGroupNodeType };
+  pathToSelection: string[];
+  specializedEntity?: string;
+  isSelected: boolean;
+  isOpened: boolean;
+}
+
+interface StatVarGroupNodeStateType {
+  renderContent: boolean;
+}
+
+export class StatVarGroupNode extends React.Component<
+  StatVarGroupNodePropType,
+  StatVarGroupNodeStateType
+> {
+  highlightedStatVar: React.RefObject<HTMLDivElement>;
+  delayTimer: NodeJS.Timeout;
+  constructor(props: StatVarGroupNodePropType) {
+    super(props);
+    this.state = {
+      renderContent: this.props.isOpened,
+    };
+    this.highlightedStatVar = React.createRef();
+    this.scrollToHighlighted = this.scrollToHighlighted.bind(this);
+  }
+
+  componentDidMount(): void {
+    this.scrollToHighlighted();
+  }
+
+  render(): JSX.Element {
+    const statVarGroup = this.props.data[this.props.statVarGroupId];
+    const triggerTitle = this.props.specializedEntity
+      ? this.props.specializedEntity
+      : statVarGroup.absoluteName;
+    const getTrigger = (opened: boolean) => {
+      return React.createElement(StatVarHierarchyNodeHeader, {
+        title: triggerTitle,
+        opened,
+        highlighted: this.props.isSelected,
+        nodeType: StatVarHierarchyNodeType.STAT_VAR_GROUP,
+      });
+    };
+    return (
+      <Collapsible
+        trigger={getTrigger(false)}
+        triggerWhenOpen={getTrigger(true)}
+        open={this.props.isOpened}
+        onOpening={() => this.setState({ renderContent: true })}
+        transitionTime={200}
+        onOpen={this.scrollToHighlighted}
+      >
+        {this.state.renderContent ? (
+          <>
+            {this.props.pathToSelection.length < 2 ? (
+              <div className="svg-node-child">
+                {this.props.data[this.props.statVarGroupId].childStatVars
+                  ? this.props.data[
+                      this.props.statVarGroupId
+                    ].childStatVars.map((statVar) => {
+                      const isSelected =
+                        this.props.pathToSelection.length === 1 &&
+                        this.props.pathToSelection[0] === statVar.id;
+                      return (
+                        <div
+                          key={statVar.id}
+                          ref={isSelected ? this.highlightedStatVar : null}
+                        >
+                          <StatVarNode
+                            dcid={this.props.dcid}
+                            statVar={statVar}
+                            nodeName={this.props.placeName}
+                            selected={isSelected}
+                          />
+                        </div>
+                      );
+                    })
+                  : null}
```

Comment:
if there isn't a need for if/else, using && also works, eg
{booleanClause && <foobarelement />}. here's a full list of options:
https://reactjs.org/docs/conditional-rendering.html

there's also a lot of nesting / render logic here. i wonder if it would be more readable to split this up into smaller components. since they're mostly for rendering, functional components could work well here (like in the first example on that page)

Reply:
Yep, agree to split this into smaller components

Reply:
done